resources/js/features/game-list/components/WantToPlayGamesDataTable/DataTableFacetedFilter.tsx

```diff
@@ -0,0 +1,186 @@
+import type { Column } from '@tanstack/react-table';
+import type { FC } from 'react';
+import { RxCheck, RxPlusCircled } from 'react-icons/rx';
+
+import { BaseBadge } from '@/common/components/+vendor/BaseBadge';
+import { BaseButton } from '@/common/components/+vendor/BaseButton';
+import {
+  BaseCommand,
+  BaseCommandEmpty,
+  BaseCommandGroup,
+  BaseCommandInput,
+  BaseCommandItem,
+  BaseCommandList,
+  BaseCommandSeparator,
+} from '@/common/components/+vendor/BaseCommand';
+import {
+  BasePopover,
+  BasePopoverContent,
+  BasePopoverTrigger,
+} from '@/common/components/+vendor/BasePopover';
+import { BaseSeparator } from '@/common/components/+vendor/BaseSeparator';
+import { buildTrackingClassNames } from '@/common/utils/buildTrackingClassNames';
+import { cn } from '@/utils/cn';
+
+interface DataTableFacetedFilterProps<TData, TValue> {
+  options: Array<{
+    label: string;
+    value: string;
+    icon?: React.ComponentType<{ className?: string }>;
+    selectedLabel?: string;
+  }>;
+
+  className?: string;
+  column?: Column<TData, TValue>;
+  isSearchable?: boolean;
+  title?: string;
+}
+
+export function DataTableFacetedFilter<TData, TValue>({
+  options,
+  column,
+  title,
+  className,
+  isSearchable = true,
+}: DataTableFacetedFilterProps<TData, TValue>) {
+  const facets = column?.getFacetedUniqueValues();
+  const selectedValues = new Set(column?.getFilterValue() as string[]);
+
+  const handleOptionToggle = (optionValue: string) => {
+    if (selectedValues.has(optionValue)) {
+      selectedValues.delete(optionValue);
+    } else {
+      selectedValues.add(optionValue);
+    }
+
+    const filterValues = Array.from(selectedValues);
+    column?.setFilterValue(filterValues.length ? filterValues : undefined);
+  };
+
+  return (
+    <BasePopover>
+      <BasePopoverTrigger asChild>
+        <BaseButton
+          size="sm"
+          className={cn(
+            'border-dashed',
```

Comment:
Let me know if this `border-dashed` appears too distracting in the UI. I'm not 100% sure about changing it because the button label mutates as the user does things (the border is supposed to hint that the button can grow or shrink).

Table borders should be a bit more subdued than the demo. If they're still too loud, I can try to tweak them some more.